src/controllers/reviews/addreview.ts

```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Response } from 'express';
+import Review from '../../database/reviews';
+import CustomError from '../../helpers';
+import { TokenRequest } from '../../middlewares/checkAuth';
+import UserData from '../../interfaces/userDataI';
+
+const addReview = async (req: TokenRequest, res: Response, next: NextFunction): Promise<void> => {
+  try {
+    const { id } = req.user as UserData;
+    const { productID, rating } = req.body;
+    await Review.create({
```

Comment:
you need to check if there is a review with same review id and user id , if its true you need to stop creating, else create review